themes/default/content/docs/intro/concepts/resources/functions.md

```diff
@@ -0,0 +1,291 @@
+</div>
+<div>
+<pulumi-choosable type="language" values="java">
+
+```java
+package generated_program;
+
+import java.util.*;
+import java.io.*;
+import java.nio.*;
+import com.pulumi.*;
+
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(App::stack);
+    }
+
+    public static void stack(Context ctx) {
+        final var example = Output.of(ApigatewayFunctions.getDomainName(GetDomainNameArgs.builder()
+            .domainName("api.example.com")
+            .build()));
+    }
+}
+```
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="python">
+
+```python
+import pulumi
+import pulumi_aws as aws
+
+example = aws.apigateway.get_domain_name(domain_name="api.example.com")
+```
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="typescript">
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+const example = pulumi.output(aws.apigateway.getDomainName({
+    domainName: "api.example.com",
+}));
+```
+
+</pulumi-choosable>
+</div>
+<div>
+<pulumi-choosable type="language" values="yaml">
+
+```yaml
+variables:
+  example:
+    Fn::Invoke:
+      Function: aws:apigateway:getDomainName
+      Arguments:
+        domainName: api.example.com
+```
+
+</pulumi-choosable>
+</div>
+</pulumi-examples></div>
+
+Provider functions are exposed in each language as a regular functions, in two variations:
+
+ 1. a function that accepts plain arguments (strings and so on) and returns a Promise, or blocks until the result is available; and,
+ 2. a function that accepts `Input` values and returns an [Output]({{< relref "/docs/intro/concepts/inputs-outputs" >}}).
+
+The documentation for a provider function will tell you the name and signature for each of the variations.
+
+### Invoke options
+
+Each function also accepts "invoke options", either as an object or as varargs depending on the host
+language. The options are as follows:
+
+| Option | Explanation                                                  |
+|--------|--------------------------------------------------------------|
+| parent | Supply a parent resource, which will be used to determine default providers, and to calculate the protect property. |
+| provider | Supply the provider to use explicitly. |
+| version | Use exactly this version of the provider plugin. |
+| pluginDownloadURL | Download the provider plugin from this URL. The download URL is otherwise inferred from the provider package. |
+| _async_ | _This option is deprecated and will be removed in a future release_ |
+
+The `parent` option has a similar purpose to the [parent option]({{< relref "/docs/intro/concepts/resources/options/parent" >}}) used when creating a resource. The parent is consulted when determining the provider to use, and the [`protect`]({{< relref "/docs/intro/concepts/resources/options/protect" >}}) status of any resources created as a result of the function invocation.
```

Comment:
Protect doesn't apply to invokes - so I believe that shouldn't be included here.

Reply:
Do you mean it's irrelevant, or not respected?

Reply:
There is nothing to protect (there's no resource being managed, so nothing to prevent deletion of), so I suppose "irrelevant".

Reply:
I was unclear on whether functions could create resources. If they respect `parent`, the suggestion is that they might; and if so, why is `protect` irrelevant? (And if not, should I remove `parent` there as well?)

Reply:
I've taken the mention of `protect` out. (The answer to my question is that `Invoke` in the engine doesn't give the callee the means to re-enter the engine, in order to e.g., create resources.)